src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -205,9 +205,16 @@ public File newFolder(String... paths) throws IOException {
             file = new File(root, relativePath.getPath());
 
             lastMkdirsCallSuccessful = file.mkdirs();
-            if (!lastMkdirsCallSuccessful && !file.isDirectory()) {
-                throw new IOException(
-                        "could not create a folder with the path \'" + relativePath.getPath() + "\'");
+            if (!lastMkdirsCallSuccessful) {
+                if (file.exists() && file.isDirectory()) {
+                    continue;
+                } else if (file.exists()) {
+                    throw new IOException(
+                            "a file with the path \'" + relativePath.getPath() + "\' exists");
```

Comment:
We can include `already` in the message, as `createNewFile` includes it in error message.

Reply:
I was thinking about the word "already" for this message two. But it is wrong here because the user wants to create a folder and not a file. Therefore we cannot say the file is already there.